### DISCUSSION TITLE
Increase header size limit of nginx directive from 8kb to 16kb

### DISCUSSION THREAD

**Discussion by rittikdasgupta** [Kong/kong#10868] [Help] [answered]
I want to increase the header size limit from 8kb to 16kb for nginx directive inside kong for a usecase. Followed the following solutions but it didn't work
- [https://docs.konghq.com/gateway/latest/reference/configuration/#nginx_http_large_client_header_buffers](https://docs.konghq.com/gateway/latest/reference/configuration/#nginx_http_large_client_header_buffers)

- [https://github.com/Kong/kong/issues/3974#issuecomment-482105126](https://github.com/Kong/kong/issues/3974#issuecomment-482105126)

Would appreciate if someone can help.

**Comment by hanshuebner** [comment]
@rittikdasgupta Can you provide some details on how it does not work?  Are headers still cut at 8kb or are you seeing some other behavior?

**Reply by rittikdasgupta** [reply, marked as answer]
This is resolved. I have set `nginx_http_large_client_header_buffers` to `4 16k` in kong's env
